chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/surveillanceReport/QuarterlyReportGenerationJob.java

```diff
@@ -0,0 +1,225 @@
+                            }
+                        }
+                    }
+                }
+            });
+        } else {
+            UserDTO user = (UserDTO) jobDataMap.get(USER_KEY);
+            if (user != null && user.getEmail() != null) {
+                sendEmail(user.getEmail(), env.getProperty("surveillance.quarterlyReport.failure.subject"),
+                        env.getProperty("surveillance.quarterlyReport.badJobData.htmlBody"), null);
+            }
+        }
+        LOGGER.info("********* Completed the Quarterly Report Generation job. *********");
+    }
+
+    private boolean isJobDataValid(JobDataMap jobDataMap) {
+        boolean isValid = true;
+        UserDTO user = (UserDTO) jobDataMap.get(USER_KEY);
+        if (user == null) {
+            isValid = false;
+            LOGGER.fatal("No user could be found in the job data.");
+        }
+
+        Long quarterlyReportId = (Long) jobDataMap.get(QUARTERLY_REPORT_ID_KEY);
+        if (quarterlyReportId == null) {
+            isValid = false;
+            LOGGER.fatal("No quarterly report ID could be found in the job data.");
+        }
+        return isValid;
+    }
+
+    private Workbook createWorkbook(QuarterlyReportDTO report) {
+        Workbook workbook = null;
+        try {
+                QuarterlyReportBuilderXlsx reportBuilder = reportBuilderFactory.getReportBuilder(report);
+                if (reportBuilder != null) {
+                    workbook = reportBuilder.buildXlsx(report);
+                } else {
+                    String msg = msgUtil.getMessage("report.quarterlySurveillance.builderNotFound");
+                    LOGGER.error(msg + " Report id " + report.getId());
+                }
+        } catch (IOException io) {
+            String msg = msgUtil.getMessage("report.quarterlySurveillance.export.builder.buildError");
+            LOGGER.error(msg, io);
+        } catch (Exception general) {
+            //catch any other type of exception
+            String msg = msgUtil.getMessage("report.annualSurveillance.export.builder.buildError");
+            LOGGER.error(msg, general);
+        }
+        return workbook;
+    }
+
+    private File writeWorkbookAsFile(QuarterlyReportDTO report, Workbook workbook) {
+        File writtenFile = null;
+        String filename = getFilename(report);
+        //write out the workbook contents to this file
+        OutputStream outputStream = null;
+        try {
+            writtenFile = File.createTempFile(filename, ".xlsx");
+            outputStream = new FileOutputStream(writtenFile);
+            LOGGER.info("Writing quarterly report file to " + writtenFile.getAbsolutePath());
+            workbook.write(outputStream);
+        } catch (Exception ex) {
+            String msg = msgUtil.getMessage("report.quarterlySurveillance.export.writeError");
+            LOGGER.error(msg, ex);
+        } finally {
+            try { outputStream.flush(); } catch (Exception ignore) {}
```

Comment:
There are some other checkstyle issues here. I think we want to touch on them, or figure out how to disable them if we don't agree with the rules, but one way or another I think we should work to not have errors reported before merging it